### PR TITLE
feat: add text area as a new component

### LINKED ISSUE #1136: Create `textarea` as a new component
This task is to create a new base component for textarea, and provide a *styled* version for the MSFT design system package.

From [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea):

> The HTML <textarea> element represents a multi-line plain-text editing control, useful when you want to allow users to enter a sizeable amount of free-form text, for example a comment on a review or feedback form.

**Prop requirements**:
For the sake of supporting certain styling requirements, we will likely need the following first class props:
- disabled
- placeholder
- readonly
- required

**Style**:
Should align closely to text field. This component should have the _same appearance_ across browsers

### PR DESCRIPTION
**Pull request by chrisdholt**
closes #1136 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-react-base/src/text-area/text-area.schema.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "description": "A Text area component's schema definition.",
+    "type": "object",
+    "id": "text-area",
+    "properties": {
+        "disabled": {
+            "title": "Disabled",
+            "type": "boolean"
+        },
+        "readOnly": {
+            "title": "Read only",
```
Comment (nalogiudice) [inline packages/fast-components-react-base/src/text-area/text-area.schema.json]: Read-only should be hyphenated. Also, may the title should be "HTML read-only attribute"? In text-field we do not make this a first-class prop... what is the reasoning of doing it here?

File: packages/fast-components-react-base/src/text-area/text-area.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+const managedClasses: TextAreaClassNameContract = {
+    textArea: "text-area-class",
+};
+
+/*
+ * Configure Enzyme
+ */
+configure({ adapter: new Adapter() });
+
+describe("text-field", (): void => {
```
Comment (nalogiudice) [inline packages/fast-components-react-base/src/text-area/text-area.spec.tsx]: typo: text-area

File: packages/fast-components-react-base/src/text-area/text-area.schema.json (hunk 1 above)
Comment (chrisdholt) [reply packages/fast-components-react-base/src/text-area/text-area.schema.json]: This property was removed and needs to just be removed from the schema.

File: packages/fast-components-react-msft/src/text-area/examples.data.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    detailData: {
+        placeholder: "Textarea placeholder text",
+    },
+    data: [
+        {
+            placeholder: "Text area placeholder text",
+        },
+        {
+            disabled: true,
+            placeholder: "Disabled text area text",
```
Comment (nalogiudice) [inline packages/fast-components-react-msft/src/text-area/examples.data.tsx]: Should we show a read-only example?

Comment (chrisdholt) [reply packages/fast-components-react-msft/src/text-area/examples.data.tsx]: Per other comment, read only needs to be *removed* from the schema

Comment (nalogiudice) [reply packages/fast-components-react-msft/src/text-area/examples.data.tsx]: Got it

File: packages/fast-components-react-base/src/text-area/text-area.schema.json (hunk 1 above)
Comment (chrisdholt) [reply packages/fast-components-react-base/src/text-area/text-area.schema.json]: removed

File: packages/fast-components-react-base/src/text-area/text-area.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+/*
+ * Configure Enzyme
+ */
+configure({ adapter: new Adapter() });
+
+describe("text-area", (): void => {
+    test("should have a displayName that matches the component name", () => {
+        expect((TextArea as any).name).toBe(TextArea.displayName);
+    });
+
```
Comment (janechu) [inline packages/fast-components-react-base/src/text-area/text-area.spec.tsx]: can you add a test that finds a `<textarea>` element

File: packages/fast-components-styles-msft/src/text-area/index.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    contrast,
+    focusVisible,
+    toPx,
+} from "@microsoft/fast-jss-utilities";
+import { get } from "lodash-es";
+import { ComponentStyles, ComponentStyleSheet } from "@microsoft/fast-jss-manager";
+import { fontWeight } from "../utilities/fonts";
+import {
+    disabledContrast,
+    ensureForegroundNormal,
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/text-area/index.ts]: ensureForegroundNormal does not get used

File: packages/fast-components-styles-msft/src/text-area/index.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    adjustContrast,
+    contrast,
+    focusVisible,
+    toPx,
+} from "@microsoft/fast-jss-utilities";
+import { get } from "lodash-es";
+import { ComponentStyles, ComponentStyleSheet } from "@microsoft/fast-jss-manager";
+import { fontWeight } from "../utilities/fonts";
+import {
+    disabledContrast,
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/text-area/index.ts]: disabledContrast does not get used

File: packages/fast-components-styles-msft/src/text-area/index.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import { applyTypeRampConfig } from "../utilities/typography";
+import { DesignSystem, withDesignSystemDefaults } from "../design-system";
+import { TextAreaClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    adjustContrast,
+    contrast,
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/text-area/index.ts]: contrast does not get used

File: packages/fast-components-styles-msft/src/text-area/index.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
+import { applyTypeRampConfig } from "../utilities/typography";
+import { DesignSystem, withDesignSystemDefaults } from "../design-system";
+import { TextAreaClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    adjustContrast,
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/text-area/index.ts]: adjustContrast does not get used

**Pull request merged**